Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -1587,69 +1582,88 @@ RcTcacheEntry TextureCacheBase::GetTexture(const int textureCacheSafetyColorSamp
     InvalidateTexture(oldest_entry);
   }
 
-  std::shared_ptr<HiresTexture> hires_tex;
+  VideoCommon::CustomTextureData* data = nullptr;
+  bool has_arbitrary_mipmaps = false;
   if (g_ActiveConfig.bHiresTextures)
   {
-    hires_tex = HiresTexture::Search(texture_info);
-
-    if (hires_tex)
+    auto hires_texture = HiresTexture::Search(texture_info);
```

Comment:
Sorry, I kinda forgot about this PR after my last comment on it...

Anyway, this still looks slightly wrong; after you leave the `if (g_ActiveConfig.bHiresTextures)` block, `hires_texture` (which is a `shared_ptr`) goes out of scope, which means that the `data` may now point to deleted data if that was the last copy of the `shared_ptr`. I don't know if this can ever happen in practice, but it seems a bit dangerous when it doesn't have to be; just put a `std::shared_ptr<HiresTexture> hires_tex;` before the `if` and then do `hires_texture = HiresTexture::Search(texture_info);`. Everything else can stay as-is.

Reply:
Great catch Admiral.  Currently the hires texture cache pointers would go out of scope if you turn them off and they were on previously.  So it can happen.